pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -876,8 +868,28 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env *model.E
 						Service:                    service,
 					}
 
-					configgen.buildSidecarOutboundListenerForPortOrUDS(node, listenerOpts, pluginParams, listenerMap,
-						virtualServices, actualWildcard)
+					// Minor optimization for Kubernetes statefulsets/headless services
+					// Instead of generating a single 0.0.0.0:Port listener, generate a listener
+					// for each podIP.
+					if bind == "" && service.Resolution == model.Passthrough &&
```

Comment:
We need a feature flag to guard this because in case of large companies with lots of pods for headless services, they may want to disable this logic.

Reply:
make sense